pkg/tracker/stamped/tracker.go

```diff
@@ -12,17 +12,17 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package tracker
+package stamped
 
-//go:generate go run -modfile ../../hack/tools/go.mod github.com/maxbrunsfeld/counterfeiter/v6 -generate
+//go:generate go run -modfile ../../../hack/tools/go.mod github.com/maxbrunsfeld/counterfeiter/v6 -generate
 
 import (
 	"github.com/go-logr/logr"
 	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 )
 
-//counterfeiter:generate . DynamicTracker
-type DynamicTracker interface {
-	Watch(log logr.Logger, obj runtime.Object, handler handler.EventHandler) error
+//counterfeiter:generate . StampedTracker
+type StampedTracker interface {
+	Watch(log logr.Logger, stampedObj runtime.Object, handler handler.EventHandler) error
```

Comment:
Given that both StampedTracker and DependencyTracker are all about .. Tracking (in the sense of ultimately getting reconciliation requests to the owner whenever there's a change to the tracked object - even though Stamped and Dependency do so in different ways), it made me wonder if the method here should be called "Track" rather than "Watch"; have you considered that?

I don't have a strong opinion here, just wondering 😁  thank you!

Reply:
I don't think we can change that one as it is creating an interface around `ObjectTracker` from an external package, used [here](https://github.com/vmware-tanzu/cartographer/blob/main/pkg/registrar/registrar.go#L114).

We could change the DependencyTracker to `watch` instead of `track`, though, if you think that's better

Reply:
aaaaah I see, makes sense
no worries, just a thought - tbh, I think `Track` makes more sense for dependency as at least for me, kubernetes-wise "watch" for me sorta implies that there'll be a `&watch=true` in a request under the hood (true for ObjectTracker where the method is indeed "Watch", not for the other)